src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -914,6 +914,49 @@
     end
   end
 
+  describe 'POST #save_prjconf' do
+    before do
+      login user
+    end
+
+    context 'can save a project config' do
+      before do
+        post :save_prjconf, params: { project: user.home_project.name, config: 'save config' }
+      end
+
+      it { expect(flash[:success]).to eq('Config successfully saved!') }
+      it { expect(response.status).to eq(200) }
+    end
+
+    context 'cannot save a project config' do
+      before do
+        allow_any_instance_of(ProjectConfigFile).to receive(:save).and_return(nil)
+        post :save_prjconf, params: { project: user.home_project.name, config: '' }
+      end
+
+      it { expect(flash[:error]).not_to be_nil }
```

Comment:
An argument against testing the error message is that it makes the test too brittle, i.e. if you have to fix a typo in the error message you shouldn't have to update an rspec test for that too. But on the other hand if you have several different kinds of flash messages which might be displayed then testing the error message in rspec is a quick win for that. 

So I think we should avoid testing the error messages if we don't have to (like if there is only one possible flash error message that could arise) but if there are multiple types of flash messages then we should test the message string.

Just my 2 cents though of course!